src/components/shared/Hooks/useContextMenu.tsx

```diff
@@ -0,0 +1,45 @@
+import { useEffect, useCallback, useState } from 'react';
+
+// from here https://dev.to/rafi993/implementing-context-menu-using-react-hooks-34ke
+const useContextMenu = (outerRef: any) => {
+  const [xPos, setXPos] = useState('0px');
+  const [yPos, setYPos] = useState('0px');
+  const [menu, showMenu] = useState(false);
+  const [cellValue, setCellValue] = useState();
+  const [cellRow, setCellRow] = useState();
+
+  const handleContextMenu = useCallback(
+    (event) => {
+      event.preventDefault();
+      if (outerRef && outerRef.current.contains(event.target)) {
+        console.log('event', event);
```

Comment:
Remember to remove the [console log](https://memegenerator.net/img/instances/68620658/a-wild-consolelog-appeared.jpg) later.